colexec: add JSONFetchValPath operator for vectorized engine

haseth opened the pull request:
Earlier JSONFetchValPath binary operator was not supported in vectorized engine. 
This PR adds support for `JSONFetchValPath (#>)`binary operator and tests it. 

Fixes: #49471 

Release note (sql change): Vectorized engine now supports JSONFetchValPath (#>) operator.

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=55570) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=55570) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=55570) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55570)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
Thanks, this almost looks good to me! I think you need to run `make execgen` to regenerate the generated code of the vectorized engine, include the changes, and then the build should be green.

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55570#-:-MJeRRzZ9cIDXkRQNnRV:b-t06oj3)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @haseth)

___
*[pkg/sql/colexec/execgen/cmd/execgen/overloads_bin.go, line 200 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/55570#-MJePjc_3FSX1V1SVRmO:-MJePjca2f3sXYJzu4rE:bj6255r) ([raw file](https://github.com/cockroachdb/cockroach/blob/fee0dd5545e799b5f1129623c4531c5aa26c8f68/pkg/sql/colexec/execgen/cmd/execgen/overloads_bin.go#L200)):*
> ```Go
> 	}
> 
> 	binOpOutputTypes[tree.JSONFetchValPath] = map[typePair]*types.T{
> ```

super nit: I'd move this declaration to be below `JSONFetchVal` to maintain the order.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
Very cool! I think there is only one last thing remaining - you need to fetch the latest master and rebase on top of it. The thing is that I fixed some issues with projection operators in #55195, and the new `JSONFetchValPath` operator should pick those up, but I don't see it in the generated code.

> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55570#-:-MJgejKIA8-kigxYUBco:bshat1k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @haseth)



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
You need to run `make execgen` again.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55570#-:-MJjRuS5DpGvCj0j2-om:b4utijp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @haseth and @yuzefovich)



<!-- Sent from Reviewable.io -->

haseth committed:
colexec: added JSONFetchValPath operator for vectorized engine

Release note (sql change): Vectorized engine now supports JSONFetchValPath (#>) operator.

yuzefovich reviewed (approved):
Thanks! <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> I'll wait for green build and will merge then.

> Reviewed 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55570#-:-MJjbDi3-8F4yitPDAH-:b2mxy6y)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

yuzefovich commented:
Thanks for your contribution!

bors r=yuzefovich
